### PR TITLE
PROJQUAY-12 - remove mirror/rules API

### PR DESCRIPTION
**Pull request by thomasmckay**
Updated API to remove mirror/rules endpoint. Rule is now handled within body of config only.

https://issues.jboss.org/browse/PROJQUAY-12

### CODE REVIEW

File: endpoints/api/mirror.py [hunk 1]
```diff
@@ -231,7 +232,7 @@ def get(self, namespace_name, repository_name):
       'sync_retries_remaining': mirror.sync_retries_remaining,
       'sync_status': mirror.sync_status.name,
       'root_rule': {
-        'rule_type': 'TAG_GLOB_CSV',
+        'rule_type': 1,
```
Comment (kurtismullins) [inline endpoints/api/mirror.py]: I believe that we should continue using `TAG_GLOB_CSV` or some human-friendly String.
The integer `1` is simply an internal enumeration and probably shouldn't be exposed through the API.

Comment (thomasmckay) [reply endpoints/api/mirror.py]: @kurtismullins other tables seem to use a "kind" that maps a string to an int. should we future proof the rules and change "rule_type" to "rule_kind" in the API? we wouldn't change the db columns but could if we wanted in the future

https://docs.quay.io/api/swagger/#!/repository/listRepos
repo_kind

Comment (thomasmckay) [reply endpoints/api/mirror.py]: Replying to myself... yes, if we're going to use strings let's stick with the `kind` pattern.

Comment (thomasmckay) [reply endpoints/api/mirror.py]: @kurtis updated

**Comment by thomasmckay** [comment]
@kurtismullins - working on the tests. please review based on the code. thanks!

**Pull request merged**